packages/aws-cdk-lib/aws-ssm/test/parameter.test.ts

```diff
@@ -652,6 +656,100 @@ test('fromLookup will use the SSM context provider to read value during synthesi
   ]);
 });
 
+test('fromLookup will persist value in context by default', async () => {
+  // GIVEN
+  const app = new cdk.App({ context: { [cxapi.NEW_STYLE_STACK_SYNTHESIS_CONTEXT]: false } });
+  const stack = new cdk.Stack(app, 'my-staq', { env: { region: 'us-east-1', account: '12344' } });
+
+  // WHEN
+  const value = ssm.StringParameter.valueFromLookup(stack, 'my-param-name');
+
+  const synth = app.synth();
+
+  const mockCloudExecutable = new MockCloudExecutable({
+    stacks: synth.stacks,
+    missing: synth.manifest.missing,
+  });
+
+  let requestedParameterName: string | undefined;
+  mockCloudExecutable.sdkProvider.stubSSM({
+    getParameter(request) {
+      requestedParameterName = request.Name;
+      return {
+        Parameter: {
+          Value: '123',
+        },
+      };
+    },
+  });
+
+  const cdkToolkit = new CdkToolkit({
+    cloudExecutable: mockCloudExecutable,
+    configuration: mockCloudExecutable.configuration,
+    sdkProvider: mockCloudExecutable.sdkProvider,
+    deployments: new Deployments({ sdkProvider: mockCloudExecutable.sdkProvider }),
+  });
+
+  await mockCloudExecutable.configuration.load();
+
+  await cdkToolkit.assembly();
+
+  await mockCloudExecutable.configuration.saveContext();
+  const config = await new Configuration({ readUserContext: false }).load();
+
+  // THEN
+  expect(value).toEqual('dummy-value-for-my-param-name');
+  expect(requestedParameterName).toEqual('my-param-name');
+  expect(config.context.keys).toEqual(['ssm:account=12344:parameterName=my-param-name:region=us-east-1']);
+});
+
+test('fromLookup will not persist value in context if requested', async () => {
```

Comment:
With `disableContextCaching` would the value in `cdk context` get overwritten when the parameter is resolved or is it the value never gets saved in `cdk context`?

If its the former I'd prefer if this test showed that with `disableContextCaching` the context will have a new value everytime it resolves the parameter

Reply:
If there's a previous value in the context, `valueFromLookup()` will use that one and not do another lookup.

This PR only allows disabling persisting to the context cache if the value needs to be looked up.

Reply:
I added a new test to confirm what I mentioned in my comment above.

Reply:
I'm open to renaming this to something like skipContextPersistance or some other name that would be more descriptive.